packages/repository/src/repositories/relation.repository.ts

```diff
@@ -100,10 +115,35 @@ export class DefaultHasManyEntityCrudRepository<
     where?: Where,
     options?: Options,
   ): Promise<number> {
-    return await this.targetRepository.updateAll(
+    return (await this.targetRepositoryGetter()).updateAll(
       constrainDataObject(dataObject, this.constraint),
       constrainWhere(where, this.constraint),
       options,
     );
   }
 }
+
+export class DefaultBelongsToEntityCrudRepository<
```

Comment:
I thought we could keep the repository pattern for potential addition of methods that a belongsTo repository could have. An example would be a method that can change what model the given model could belong to.

Reply:
> I thought we could keep the repository pattern for potential addition of methods that a belongsTo repository could have. An example would be a method that can change what model the given model could belong to.

Makes sense. I agree that a repository is easier to extend than the current `Promise<Target`> API.

In that case, let's follow the plan B outlined in my comment above:
 - mark it as implements `BelongsToRepository<...>`
 - modify `BelongsToFactory` to return `BelongsToRepository<...>` instead of `Promise<Target>`.

Thoughts?

I hope I am not missing something important here.

Reply:
If we were to implement your suggestion, the controller method for finding the related model would look something like this:
```ts
findCustomer(id: number) {
  return await orderRepo.customer(id).find();
  // instead of: await orderRepo.customer(id);
}
```

I know that @raymondfeng and @virkt25 like `BelongsToFactory` returning `Promise<Target>` here, but I can also see that we probably don't want the API changing too drastically in the future.
Thoughts?

Reply:
I am personally find with both flavors. Both of them have pros and cons, I think we need to make a trade-off here.

- `orderRepo.customer(orderId)`
- `orderRepo.customer(orderId).find`.

Can we rename the repository method from `find` to `get` please?

```ts
findCustomer(id: number) {
  return await orderRepo.customer(id).get();
}
```

Reply:
I've decided to go with the `orderRepo.customer(orderId)`, primarily because it requires the least amount of deviation from our currently established pattern with `DefaultHasManyEntityCrudRepository` where we pass in a 'constraint' to the relation repository.